geps/gep-1494/index.md

```diff
@@ -130,20 +130,383 @@ From @ongy, some additional goals to keep in mind:
+  Policy later, must not override explicit config - that would
+  violate one of the key declarative principles, that what is requested in the
+  spec is either what ends up in the state, or that config is rejected.
+* Secondly, filter ordering is particularly important for Auth use cases, so we
+  must ensure that when we add Policy defaulting we have a way to indicate at
+  what position in a filter list the Auth policy should fit.
+
+### Why two phases?
+
+In short: In the interest of getting something, even if incomplete, into our
+user's hands as quickly as possible.
+
+Policy design is complex, and needs to be done carefully. Doing a first
+pass using only a Filter to get the basic config correct while we discuss
+how to make the Policy handling work means that we can get some work out to the
+community without needing to complete the whole design.
+
+In particular, the design for the Filter plus Policy will need to answer at
+least the following questions:
+
+* How to set where in a list of Filters a defaulted Auth filter sits;
+  and what happens if no Filters are specified in a HTTPRoute? Does it go first,
+  last, or should there be a way to specify the order?
+* What Policy types are possible? Defaults? (Definitely useful for setting a
+  baseline expectation for a cluster, which is desirable for security constructs
+  like Auth) Overrides? (Also very useful for ensuring that exceptions meet
+  certain requirements - like only allowing the disabling of Auth on `/healthz`
+  endpoints or similar use cases.)
+* Should Policy have a way to describe rules around when it should take effect?
+  That's in addition to the usual hierarchical rules, should the Policy have ways
+  to include or exclude specific matches? This would require agreement in the
+  Policy Attachment spec as well.
+
+All of these changes have costs in complexity and troubleshooting difficulty, so
+it's important to ensure that the design consciously makes these tradeoffs.
+
+In particular, the last two items in the above list seem likely to require a fair
+amount of discussion, and including a Policy in the initial release of this
+seems likely to make this change miss its current release window.
+
+
+### Why Envoy's ext_auth?
+
+#### What is ext_auth?
+
+Envoy's External Authorization filter is a filter that calls out to an authorization
+service to check if the incoming request is authorized or not. Note that, in
+order to check _authorization_, it must also be able to determine _authentication_ - 
+this is one of the reasons why we've chosen this approach.
+
+Envoy's implementation of this filter allows both a
+[gRPC, protobuf API](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_authz/v3/ext_authz.proto)
+and configuration of a HTTP based API (which, as it's not defined using a
+specification like protobuf, requires more configuration).
+
+The important thing to remember here is that the actual authentication and
```

Comment:
This is partially true - authentication can be done in addition to authorization by the external server, but the external server can also rely on proxy authentication from earlier stages, e.g. Envoy JWT filter, frontend mTLS identity etc.

Reply:
Updated, thanks.